quickwit/quickwit-indexing/src/source/file_source.rs

```diff
@@ -140,11 +141,14 @@ impl TypedSourceFactory for FileSourceFactory {
         let mut offset = 0;
         let reader: Box<dyn AsyncRead + Send + Unpin> = if let Some(filepath) = &params.filepath {
             let partition_id = PartitionId::from(filepath.to_string_lossy().to_string());
-            if let Some(Position::Offset(offset_str)) =
-                checkpoint.position_for_partition(&partition_id).cloned()
-            {
-                offset = offset_str.parse::<usize>()?;
-            }
+            offset = checkpoint
+                .position_for_partition(&partition_id)
+                .map(|position| {
```

Comment:
I think we should log an error and act as if we did not have a position for this partition.